src/read.c

```diff
@@ -1175,11 +1181,19 @@ void ReadFuncExpr (
 	    ASS_LIST( nams, narg+nloc, name );
 	    Match(S_IDENT,"identifier",S_RPAREN|S_LOCAL|STATBEGIN|S_END|follow);
 	}
+
+	if(TLS(Symbol) == S_DOTDOT) {
+	    SyntaxError("Three dots required for variadic argument list");
+	}
+	if(TLS(Symbol) == S_DOTDOTDOT) {
+	    isvarg = 1;
+	    GetSymbol();
+    }
+
 	while ( TLS(Symbol) == S_COMMA ) {
-	    if (narg > 0 && !strcmp(CSTR_STRING(ELM_LIST(nams,narg)),"arg"))
-	      {
-		SyntaxWarning("arg used not as the last argument");
-	      }
+	    if (isvarg) {
+	        SyntaxError("Only final argument can be variadic");
+	    }
```

Comment:
Very minor point: Right now, one gets this:
```
gap> function(x..,y) end;
Syntax error: Three dots required for variadic argument list
function(x..,y) end;
           ^
```
That might be slightly confusing / misleading, so can we change this to print "only the final argument can be variadic" instead? If not resp. if this is a lot of work, then just ignore this -- I am merely wondering≥